### PR TITLE
add_script_s3_bucket_deletion

### PR DESCRIPTION
**Pull request by ILpinto**
The main purpose of this script is to delete a bucket associated with a cluster
after the cluster is uninstalled

### CODE REVIEW

File: aws/delete_s3_velero_bucket.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+
+def delete_velero_cluster_buckets(cluster: str) -> None:
+    """
+    Delete the velero bucket associated with a cluster
+
+    Args:
+        cluster: The cluster name
+
+    Returns:
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:60]: No need to set `Returns` in docstring if the function do not retirn anything

File: aws/delete_s3_velero_bucket.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+def verify_aws_env_requirements_met() -> None:
+
+    """
+    Verify that the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are set.
+    The script exits if neither is set.
+
+    Returns: None
+    """
+    require_env = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"]
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:290]: you also need to check `S3_CLUSTER` right? or remove line 14

File: aws/delete_s3_velero_bucket.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    for env_var in require_env:
+        if not os.getenv(env_var):
+            LOGGER.error(
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:293]: I would collect all the env that not set and log one error with the missing ones

File: aws/delete_s3_velero_bucket.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+def get_cluster_name() -> str:
+    """
+    Get the cluster name passed to the script
+
+    The cluster name can be passed as a command line argument or an environment variable.
+    The script exits if the cluster name is not provided
+
+    Args:
+        None
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:254]: Please remove all things that have `None` (`Args` in this case)

File: aws/delete_s3_velero_bucket.py [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+        """,
+        default=os.getenv("S3_CLUSTER"),
+    )
+
+    cluster_name = parser.parse_args().cluster_name
+    if cluster_name:
+        return cluster_name
+
+    LOGGER.error(
+        """
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:274]: Why set the string with `"""` ?

File: aws/delete_s3_velero_bucket.py [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+import argparse
+import json
+import os
+import re
+import sys
+from http import HTTPStatus
+from typing import Union
+
+import boto3
+from ocp_utilities.logger import get_logger
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:39]: Please set your own logger, do not use other modules logger

**Review by myakove — CHANGES_REQUESTED** [review]

File: aws/delete_s3_velero_bucket.py [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+    if cluster_name:
+        return cluster_name
+
+    LOGGER.error(
+        """
+            Required cluster value was not set!
+            Run script with --help for specifics
+        """,
+    )
+    sys.exit(1)
```
[the remark this answers is not in the document]
Comment (ILpinto) [reply aws/delete_s3_velero_bucket.py:253]: refactor and add new method verify_env_variables_and_parameters

**Comment by ILpinto** [comment]
Moved https://github.com/RedHatQE/cloud-tools/pull/4

**Pull request closed without merging**